community/kernel/src/main/java/org/neo4j/kernel/api/exceptions/schema/DuplicateNodeSchemaRuleException.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright (c) 2002-2015 "Neo Technology,"
+ * Network Engine for Objects in Lund AB [http://neotechnology.com]
+ *
+ * This file is part of Neo4j.
+ *
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package org.neo4j.kernel.api.exceptions.schema;
+
+import org.neo4j.kernel.api.TokenNameLookup;
+
+import static java.lang.String.format;
+
+public class DuplicateNodeSchemaRuleException extends DuplicateSchemaRuleException
```

Comment:
This and DuplicateNodeSchemaRuleException could be merge together and accept an `EntityType` to distinguish between the two. I can't imagine one handling node/relationship case differently. Also many other exceptions do this by passing in `EntityType`